okapi-ir/src/test/scala/org/opencypher/okapi/ir/impl/typer/SchemaTyperTest.scala

```diff
@@ -40,417 +40,403 @@ import org.scalatest.mockito.MockitoSugar
+class SchemaTyperTest extends SchemaTyperTestSuite with MockitoSugar {
+
+  val baseTypes = Seq[CypherType](
+    CTAny, CTNumber, CTNull, CTVoid,
+    CTBoolean, CTInteger, CTFloat, CTString,
+    CTDate, CTLocalDateTime, CTDuration,
+    CTIdentity, CTPath
+  )
+
+  val simple =
+    baseTypes.map(tpe => tpe.name -> tpe) ++
+    baseTypes.map(tpe => s"${tpe.name}_OR_NULL" -> tpe.nullable)
+
+  private val lists = simple
+    .map { case (n, t) => s"LIST_$n" -> CTList(t) }
+
+  private val maps = Seq(
+    "NODE" -> CTNode(Set("Node")),
+    "NODE_EMPTY" -> CTNode(),
+    "REL" -> CTRelationship(Set("REL")),
+    "REL_EMPTY" -> CTRelationship(),
+    "MAP" -> CTMap(simple.toMap),
+    "MAP_EMPTY" -> CTMap(Map())
+  )
+
+  private val all = Seq(
+    simple,
+    lists,
+    lists.map { case (n, t) => s"${n}_OR_NULL" -> t.nullable },
+    maps,
+    maps.map { case (n, t) => s"${n}_OR_NULL" -> t.nullable }
+  ).flatten
+
+  private val properties  =
+    simple ++ Seq("name" -> CTString,  "age" -> CTInteger)
+
+  private val properties2 =
+    simple ++ Seq("name" -> CTBoolean, "age" -> CTFloat)
+
+  private val propertiesJoined =
+    simple ++ Seq("name" -> CTAny, "age" -> CTNumber)
+
+  private val schema: Schema = Schema.empty
+    .withNodePropertyKeys("Node")(properties : _*)
+    .withRelationshipPropertyKeys("REL")(properties: _*)
+    .withNodePropertyKeys("Node2")(properties2 : _*)
+    .withRelationshipPropertyKeys("REL2")(properties2: _*)
+
+  implicit val typer: SchemaTyper =
+    SchemaTyper(schema)
+
+  implicit val context: TypeTracker =
+    typeTracker(all: _*)
+      .withParameters(Map(
+        "STRING_EMPTY" -> CypherValue(""),
+        "STRING_BOOLEAN" -> CypherValue("BOOLEAN"),
+        "FLOAT_0" -> CypherValue(0.0),
+        "INTEGER_0" -> CypherValue(0),
+        "BOOLEAN_FALSE" -> CypherValue(false),
+        "NULL" -> CypherValue(null)
+      ))
 
   it("should report good error on unsupported functions") {
-    implicit val context: TypeTracker = typeTracker("a" -> CTList(CTInteger))
+    val f = Tail.asInvocation(Variable("LIST_INTEGER")(pos))(pos)
 
-    val f = Tail.asInvocation(Variable("a")(pos))(pos)
-
-    assertExpr.from("tail(a)") shouldFailToInferTypeWithErrors(
+    assertExpr.from("tail(LIST_INTEGER)") shouldFailToInferTypeWithErrors(
       UnsupportedExpr(f), NoSuitableSignatureForExpr(f, Seq(CTList(CTInteger)))
     )
   }
 
   it("should type Date") {
-    implicit val context: TypeTracker = typeTracker("d" -> CTDate)
-
-    assertExpr.from("date()") shouldHaveInferredType CTDate.nullable
-    assertExpr.from("date('2010-12-10')") shouldHaveInferredType CTDate.nullable
-    assertExpr.from("date({ year: 2018, month: 12, day: 18 })") shouldHaveInferredType CTDate.nullable
+    assertExpr.from("date()") shouldHaveInferredType CTDate
+    assertExpr.from("date(STRING)") shouldHaveInferredType CTDate
+    assertExpr.from("date({ year: 2018, month: 12, day: 18 })") shouldHaveInferredType CTDate
```

Comment:
Should it be like this?
```suggestion
    assertExpr.from("date(MAP)") shouldHaveInferredType CTDate
```